File: 04-gridpulse-ev-transformer/sensors/mqttPublisher.js
'use strict';

// contract: gridpulse/{hubId}/{category}/{metric}, body is the reading JSON verbatim
function publishReading(mqttClient, reading) {
  const topic = `gridpulse/${reading.hubId}/${reading.category}/${reading.metric}`;
  mqttClient.publish(topic, JSON.stringify(reading));
}

module.exports = { publishReading };

File: 15-harborpulse-vessel-monitoring/sensors/mqtt_publisher.py
"""Publishes sensor readings onto the harborpulse/{vesselId}/{metric} MQTT topic contract."""
import json

TOPIC_TEMPLATE = "harborpulse/{vesselId}/{metric}"


def publish_reading(mqtt_client, reading: dict) -> None:
    topic = TOPIC_TEMPLATE.format(vesselId=reading["vesselId"], metric=reading["metric"])
    payload = json.dumps(reading)
    mqtt_client.publish(topic, payload)
